front/lib/api/files/preprocessing.ts

```diff
@@ -168,6 +171,52 @@ const extractTextFromPDF: PreprocessingFunction = async (
   }
 };
 
+// CSV preprocessing.
+// We upload the content of the CSV on the processed bucket and the schema in the snippet bucket.
+const extractContentAndSchemaFromCSV: PreprocessingFunction = async (
+  auth: Authenticator,
+  file: FileResource
+): Promise<Result<undefined, Error>> => {
+  try {
+    const readStreamForProcessedFile = file.getReadStream(auth, "original");
+    const processedWriteStream = file.getWriteStream(auth, "processed");
+    await pipeline(readStreamForProcessedFile, processedWriteStream);
```

Comment:
Can we call `storeRawText` instead of duplicating the logic?

Or if we want to handle big CSV files, I'd suggest that we use `PassThrough` so we create one read stream and send the data to many listeners.

Reply:
Ok I tried something and it seems it's working!